src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertions.php

```diff
@@ -0,0 +1,413 @@
+
+        $inputNode = static::$client->getCrawler()->filter("input[name=\"$fieldName\"]");
+
+        Assert::assertGreaterThan(0, $inputNode->count(), "Input with name [$fieldName] not found on the page.");
+
+        Assert::assertNotEquals(
+            $expectedValue,
+            $value = $inputNode->getNode(0)->getAttribute('value'),
+            "Expected value [$value] for the [$fieldName] input was expected to not equal the actual value [$value].");
+    }
+
+    public static function assertRouteEquals($expectedRoute, array $parameters = []): void
+    {
+        static::checkRequestAvailable();
+
+        Assert::assertSame(
+            $expectedRoute,
+            $route = static::$client->getRequest()->attributes->get('_route'),
+            "Expected route name [$expectedRoute] does not match the actual value [$route]."
+        );
+
+        if (\count($parameters)) {
+            foreach ($parameters as $key => $expectedValue) {
+                static::assertRequestAttributeValueEquals($key, $expectedValue);
+            }
+        }
+    }
+
+    public static function assertRequestAttributeValueEquals(string $key, $expectedValue): void
+    {
+        Assert::assertSame(
+            $expectedValue,
+            $value = static::$client->getRequest()->attributes->get($key),
+            "Expected request attribute [$key] value [$expectedValue] does not match actual value [$value]."
+        );
+    }
+
+    protected static function getResponseCookieFromClient(string $name, string $path = '/', string $domain = null): ?Cookie
+    {
+        static::checkResponseAvailable();
+
+        $cookies = static::$client->getResponse()->headers->getCookies();
+
+        $filteredCookies = \array_filter($cookies, function (Cookie $cookie) use ($name, $path, $domain) {
+            return
+                $cookie->getName() === $name
+                && $cookie->getPath() === $path
+                && $cookie->getDomain() === $domain
+            ;
+        });
+
+        return \reset($filteredCookies) ?: null;
+    }
+
+    private static function checkClientAvailable(): void
+    {
+        if (!static::$client instanceof Client) {
+            static::fail(\sprintf(
+                'A client must be set to make assertions on it. Did you forget to call "%s::%s"?',
+                static::class, 'createClient'
+            ));
+        }
+    }
+
+    private static function checkCrawlerAvailable(): void
+    {
+        static::checkClientAvailable();
+
+        if (!static::$client->getCrawler()) {
+            static::fail('A client must have a crawler to make assertions. Did you forget to make an HTTP request?');
+        }
+    }
+
+    private static function checkResponseAvailable(): void
+    {
+        static::checkClientAvailable();
+
+        if (!static::$client->getResponse()) {
+            static::fail('A client must have an HTTP Response to make assertions.');
+        }
+    }
```

Comment:
Maybe your `checkResponseAvailable()` could be more usefull by return the response than void?

As you do `static::$client->getResponse()` every time after `checkResponseAvailable()` call

Reply:
I removed the "check" methods and just use getters 👍